lib/Token/Token.cpp

```diff
@@ -223,6 +229,12 @@ bool Token::isBasePointerType() const {
   return false;
 }
 
+bool Token::isFieldPointerType() const {
+ if (this->Ty->isPointerTy())
```

Comment:
I think this will return the type of the variable and not the field, for example, for variables like a, it will return the type of a and for of type a[1] it will again return the type of a. What do you think?

Reply:
I have changed the name of this function now. The function returns the type of the object invoking it.